datafusion/optimizer/src/test/mod.rs

```diff
@@ -152,20 +152,16 @@ pub fn assert_analyzer_check_err(
 }
 pub fn assert_optimized_plan_eq(
     rule: Arc<dyn OptimizerRule + Send + Sync>,
-    plan: &LogicalPlan,
+    plan: LogicalPlan,
     expected: &str,
 ) -> Result<()> {
-    let optimizer = Optimizer::with_rules(vec![rule.clone()]);
-    let optimized_plan = optimizer
-        .optimize_recursively(
-            optimizer.rules.first().unwrap(),
-            plan,
-            &OptimizerContext::new(),
-        )?
-        .unwrap_or_else(|| plan.clone());
+    fn observe(_plan: &LogicalPlan, _rule: &dyn OptimizerRule) {}
+
+    // in tests we are applying only one rule once
+    let opt_context = OptimizerContext::new().with_max_passes(1);
```

Comment:
it turns out if you apply this rule multiple times, it changes the output. I explicitly made the test only run it a single time to match the previous behavior